tests/jballerina-unit-test/src/test/resources/test-src/record/readonly_record_fields.bal

```diff
@@ -918,6 +918,61 @@ function testDefaultValueFromCETBeingUsedWithReadOnlyFieldsInTheMappingConstruct
     assertFalse(f is record {|readonly string a; readonly string b; readonly string[] c;|});
 }
 
+type R1 record {|
+    never x?;
+|};
+
+type R2 record {
+    never x?;
+};
+
+type R3 record {|
+    never x?;
+    readonly int y;
+|};
+
+type R4 record {|
+    never x?;
+    string y;
+|};
+
+type R5 record {|
+   record {|
+       never a;
+   |} x?;
+|};
+
+type R6 record {|
+    never|never x?;
+|};
+
+function testRecordReadonlynessWithNeverFields() {
+    record {|
+        never x?;
+        never y?;
+    |} c = {};
+
+    readonly d = c;
+    R1 e = {};
+    readonly e1 = e;
+    R2 f = {};
+    R3 g = {y: 1};
+    readonly g1 = g;
+    R4 h = {y: "abc"};
+    R5 i = {};
+    readonly i1 = i;
+    R6 j = {};
+    readonly j1 = j;
+
+    assertTrue(d is record {|never x?; never y?;|} & readonly);
```

Comment:
Reorganize so that variables and declarations are together. Helps with readability.

Reply:
Fixed.

Reply:
I think what @MaryamZi meant was something like this. Since relevant declarations and assertions are together it improves the readability.

```bal
record {|
        never x?;
        never y?;
    |} c = {};
readonly c1 = c;
assertTrue(d is record {|never x?; never y?;|} & readonly);

R1 e = {};
readonly e1 = e;
assertTrue(e1 is record {|never x?;|} & readonly);
```

Let's change d to c1 similar to how we have assigned rest.